thamos/lib.py

```diff
@@ -321,11 +323,14 @@ def advise(
     if recommendation_type:
         recommendation_type = recommendation_type.lower()
 
+    if _is_s2i():
```

Comment:
Maybe:

```python
if source_type is None and _is_s2i():
        source_type = ThothAdviserIntegrationEnum.S2I
```

.. not to override parameters, these are usually hard to debug later on.